mdbenchmark/plot.py

```diff
@@ -0,0 +1,210 @@
+        ls='--',
+        color=color,
+        alpha=.5)
+    return ax
+
+
+def plot_over_group(df, plot_nodes, ax=None):
+    # plot all lines
+    if plot_nodes is True:
+        df_sel = "nodes"
+    else:
+        df_sel = "ncores"
+    gb = df.groupby(['gpu', 'module', 'host'])
+    groupby = ['gpu', 'module', 'host']
+    for key, df in gb:
+        label = ' '.join(['{}={}'.format(n, v) for n, v in zip(groupby, key)])
+        plot_line(df=df, df_sel=df_sel, ax=ax, label=label)
+
+    # style axes
+    if plot_nodes is True:
+        ax.set_xlabel('Number of Nodes')
+    elif plot_nodes is False:
+        ax.set_xlabel('Number of Cores')
+
+
+    ax.set_ylabel('Performance [ns/day]')
+    ax.legend()
+
+    ax2 = ax.twiny()
+    ax1_xticks = ax.get_xticks()
+    ax2.set_xticks(ax1_xticks)
+    ax2.set_xticklabels(gb.get_group(list(gb.groups.keys())[0])[df_sel])
+    ax2.set_xbound(ax.get_xbound())
+    #ax2.set_xlabel('{}'.format('{}\n\nCores'.format(df['host'][0])))
+
+    return ax
+
+
+@cli.command()
+@click.option(
+    '--csv',
+    help='name of csv file',
+    multiple=True,
+    show_default=True)
+@click.option(
+    '--output-name',
+    '-o',
+    default=None,
+    help="name of output files",
+    show_default=True)
+@click.option(
+    '--output-type',
+    '-t',
+    help="file extension for plot outputs",
+    type=click.Choice(['png', 'pdf', 'svg', 'jpeg']),
+    show_default=True,
+    default='png')
+@click.option(
+    '--host-name',
+    '-h',
+    default=None,
+    multiple=True,
+    help="hostname",
+    show_default=True)
+@click.option(
+    '--module-name',
+    '-h',
+    default=None,
+    multiple=True,
+    help="module name or engine name (gromacs, namd )",
+    show_default=True)
+@click.option(
+    '--host-name',
+    '-h',
+    default=None,
+    multiple=True,
+    help="hostname",
+    show_default=True)
+@click.option(
+    '--gpu/--no-gpu',
+    help="plot data for GPU runs",
+    show_default=True,
+    default=False)
+@click.option(
+    '--cpu/--no-cpu',
+    help="plot data for GPU runs",
+    show_default=True,
+    default=True)
+@click.option(
+    '--plot-nodes/--plot-cores',
+    help="Plot performance per node instead of core",
+    show_default=True,
+    default=False)
+def plot(csv, output_name, output_type, host_name, module_name, gpu, cpu, plot_nodes):
+    """Plot nice things"""
+    print(plot_nodes)
+    df_list = []
+    for c in csv:
+        tmp_df = pd.read_csv(c, index_col=0)
+        # append df_list
+        df_list.append(tmp_df)
+
+    df = pd.concat(df_list)
+    # Remove NaN values. These are missing ncores/performance data.
+    df = df.dropna()
+
+    # preprocess the commandline entries
+
+    df_module_list = df['module'].tolist()
+    processed_module_names = []
+    print(df_module_list)
+    for module in module_name:
+        if module in ['namd', 'gromacs']:
+            real_module_names = [s for s in df_module_list if module in s]
```

Comment:
What exactly are you trying to check here?

Reply:
this is the check which is envoked when the user only gives an engine name. Then all modules which contain this engine name are used for the module selection e.g gromacs/5.1.4 gromacs/2016 would both be selected

Reply:
i would like to give the option to only give "gromacs" as option to get a plot with all gromacs version which are in a df. One would need some kind of string match to do this